modules/ding_wayf/ding_wayf.module

```diff
@@ -124,6 +124,81 @@ function ding_wayf_init() {
   }
 }
 
+
+/**
+ * Do a check on the checksum given from gatewayf.
+ *
+ * @param array $attributes
+ *   The attributes from gatewayf.
+ *
+ * @return bool
+ *   Safe or not.
+ */
+function ding_wayf_security_check(array $attributes) {
+  // Check attributes.
+  if (empty($attributes['schacPersonalUniqueID']) || empty($attributes['eduPersonTargetedID'])) {
+    // We expect alway to get these two attributes.
+    watchdog('ding_gatewayf', 'Neccesary attributes missing', array(), WATCHDOG_ERROR);
+    drupal_set_message('Sorry login failed - contact site administrator', 'warning');
```

Comment:
Why set a message here? We return a boolean and let the callee determine how to handle this.

As I see it a message will be printed on L216.